backend/hct_mis_api/apps/household/celery_tasks.py

```diff
@@ -64,26 +67,31 @@ def recalculate_population_fields_chunk_task(households_ids: List[UUID]) -> None
 @app.task()
 @log_start_and_end
 @sentry_tags
-def recalculate_population_fields_task(household_ids: Optional[List[str]] = None) -> None:
+def recalculate_population_fields_task(
+    household_ids: Optional[List[str]] = None, program_id: Optional[str] = None
+) -> None:
     from hct_mis_api.apps.household.models import Household
 
     params = {}
     if household_ids:
         params["pk__in"] = household_ids
+    recalculate_composition = None
+    if program_id:
+        program = Program.objects.get(id=program_id)
+        recalculate_composition = program.data_collecting_type.recalculate_composition
+    queryset = Household.objects.filter(**params).only("pk").order_by("pk")
+    if recalculate_composition is None:
+        queryset = queryset.filter(collect_individual_data__in=(COLLECT_TYPE_FULL, COLLECT_TYPE_PARTIAL))
+    elif not recalculate_composition:
```

Comment:
this covers what? recalculate_composition === blank "" ?

Reply:
False

Reply:
aaa ok 👍